lib/translator/read.cc

```diff
@@ -198,4 +198,13 @@ StatusCode Read16ToNvme(absl::Span<const uint8_t> scsi_cmd,
               nvme_cmd, allocation, nsid);
 }
 
+void ReadToScsi(absl::Span<uint8_t> buffer,
+                const nvme::GenericQueueEntryCmd& nvme_cmd) {
+  uint64_t data_ptr = nvme_cmd.dptr.prp.prp1;
```

Comment:
Do we have any functions to parse prp lists if the data is bigger than a single page?

Reply:
Discussed this offline but it seems like the NVMe driver is responsible for managing the prp list if there are multiple pages of data. This is supported by how the nvme cli handles the buffer https://github.com/linux-nvme/nvme-cli/blob/master/nvme.c#L4094. All that we need to do on our end is to allocate a sufficiently large buffer, set the prp pointer to the pointer to the buffer, and then the buffer will have the response data copied into it. So when translating back to SCSI, we can just copy the data pointed to by the prp to the scsi data in buffer.